vector/src/test/java/im/vector/app/features/onboarding/OnboardingViewModelTest.kt

```diff
@@ -109,81 +119,156 @@ class OnboardingViewModelTest {
     }
 
     @Test
-    fun `given homeserver does not support personalisation when registering account then updates state and emits account created event`() = runBlockingTest {
-        fakeSession.fakeHomeServerCapabilitiesService.givenCapabilities(HomeServerCapabilities(canChangeDisplayName = false, canChangeAvatar = false))
-        givenSuccessfullyCreatesAccount()
+    fun `when handling SignUp then sets sign mode to sign up and starts registration`() = runBlockingTest {
+        givenRegistrationResultFor(RegisterAction.StartRegistration, ANY_CONTINUING_REGISTRATION_RESULT)
         val test = viewModel.test(this)
 
-        viewModel.handle(OnboardingAction.RegisterDummy)
+        viewModel.handle(OnboardingAction.UpdateSignMode(SignMode.SignUp))
 
         test
-                .assertStates(
+                .assertStatesWithPrevious(
```

Comment:
this name was tricky, totally open to ideas

the aim is for each `vararg` to be given the previously expected value to assert against, allows the assertions to be focus on the differences between each state emission rather than having to pass the entire state each time

and yep the order is also asserted against as this is an exact check, any extra or missing states will fail the test 

```kotlin
val flow = flowOf {
  emit(0)
  emit(10)
  emit(12)
  emit(20)
}

flow.assertStatesWithPrevious(
  0,
  { initial -> initial + 10 },
  { next -> next + 2 },
  { next ->  next + 8 }
)
```

Reply:
Okay, great! Thanks for explaining! What about renaming to `assertStatesChanges` or `assertStatesModification` or do you prefer to keep `assertStatesWithPrevious`. In any case, I think you can add your explanation and the example as a documentation method, it helps a lot.

Reply:
`assertStatesChanges` sounds good! will also add some documentation